src/sliceDataStorage.cpp

```diff
@@ -22,6 +22,39 @@ Polygons& SliceLayerPart::getOwnInfillArea()
     }
 }
 
+bool SliceLayerPart::isUsed(const SettingsBaseVirtual& mesh_settings)
+{
+    if (mesh_settings.getSettingAsCount("wall_line_count") > 0 && insets.size() > 0)
+    { // note that in case wall line count is zero, the outline was pushed onto the insets
+        return true;
+    }
+    if (skin_parts.size() > 0)
+    {
+        return true;
+    }
+    if (mesh_settings.getSettingBoolean("spaghetti_infill_enabled"))
+    {
+        if (spaghetti_infill_volumes.size() > 0)
+        {
+            return true;
+        }
+    }
+    else if (mesh_settings.getSettingInMicrons("infill_line_distance") > 0)
+    {
+        for (const std::vector<Polygons>& infill_area_per_combine : infill_area_per_combine_per_density)
+        {
+            for (const Polygons& area : infill_area_per_combine)
+            {
+                if (area.size() > 0)
+                {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
```

Comment:
It might be worth considering if turning these if-statements around could make the code simpler: E.g. when infill density is 0, return `false`, if spaghetti is enabled but the volumes are empty, return `false`, etc. Perhaps that makes it read a bit more easily.

(Not blocking. And I don't know if that would make it better.)